zstacklib/zstacklib/utils/lru_cache.py

```
@@ -0,0 +1,84 @@
+        def _cache_wrapper(self, caller, *args, **kwargs):
+            # Create a unique key including the types (in order to differentiate between 1 and '1'):
+            kwargs_key = "".join(map(lambda x : str(x) + str(type(kwargs[x])) + str(kwargs[x]), sorted(kwargs)))
+            key = "".join(map(lambda x : str(type(x)) + str(x) , args)) + kwargs_key
+
+            # Check if caller exists, if not create one:
+            if caller not in self._caches_dict:
+                self._caches_dict[caller] = [collections.OrderedDict(), time.time()]
+            else:
+                # Validate in case the refresh time has passed:
+                if self._timeout != None:
+                    if time.time() - self._caches_dict[caller][1] > self._timeout:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)